### PR TITLE
Make OnDrag, OnDragEnd, and OnMouseUp return void instead of bool

### PR DESCRIPTION
**Pull request by smoogipoo**
While working on a related fix, I found an issue relating to mouse input whereby a hierarchy that received mouse-down events may not receive mouse-up events if a drawable above them blocked it.

This isn't limited to `OnMouseUp`, but all mouse-release events (`OnMouseUp`, `OnDrag`, `OnDragEnd`), for lack of better naming.

I'm doing this only for mouse events for now since they're the only input events that are correctly. Key, joystick, and keybinding events are all broken in other ways (I'm working on fixing them). Those other events will be changed after said fixes have been applied.

Here's an example illustrating this, although this isn't really testable anymore (hence why I haven't added a test):

```csharp
[Test]
public void DoTest()
{
    var receptors = new InputReceptor[3];

    AddStep("setup", () =>
    {
        Children = new Drawable[]
        {
            receptors[0] = new InputReceptor
            {
                Size = new Vector2(100),
                MouseDown = () => true,
            },
            receptors[1] = new InputReceptor { Size = new Vector2(100) },
            receptors[2] = new InputReceptor
            {
                Size = new Vector2(100),
                MouseUp = () => true,
            }
        };
    });

    AddStep("move mouse to receptors", () => InputManager.MoveMouseTo(receptors[0]));
    AddStep("mousedown", () => InputManager.PressButton(MouseButton.Left));
    AddStep("mouseup", () => InputManager.ReleaseButton(MouseButton.Left));

    AddAssert("all received down", () => receptors.All(r => r.DownReceived)); // Passes
    AddAssert("0 received up", () => receptors[0].UpReceived); // Fails
    AddAssert("1 received up", () => receptors[1].UpReceived); // Fails
    AddAssert("2 received up", () => receptors[2].UpReceived); // Passes
}

private class InputReceptor : Box
{
    public bool DownReceived;
    public bool UpReceived;

    public Func<bool> MouseDown;
    public Func<bool> MouseUp;

    protected override bool OnMouseDown(MouseDownEvent e)
    {
        DownReceived = true;
        return MouseDown?.Invoke() ?? false;
    }

    protected override bool OnMouseUp(MouseUpEvent e)
    {
        UpReceived = true;
        return MouseUp?.Invoke() ?? false;
    }
}
```

# Breaking changes:

# vNext

## `OnDrag`, `OnDragEnd`, and `OnMouseUp` now return `void`

These events can no longer be blocked by other `Drawable`s in the hierarchy.

This guarantees that `Drawable`s that have received the respective `OnDragStart` and `OnMouseDown` events will receive accompanying `OnDrag`, `OnDragEnd` and `OnMouseUp` events

### CODE REVIEW

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -1932,12 +1935,12 @@ protected virtual void OnHoverLost(HoverLostEvent e)
         }
 
         protected virtual bool OnMouseDown(MouseDownEvent e) => Handle(e);
-        protected virtual bool OnMouseUp(MouseUpEvent e) => Handle(e);
+        protected virtual void OnMouseUp(MouseUpEvent e) => Handle(e);
```
Comment (iiSaLMaN) [inline osu.Framework/Graphics/Drawable.cs]: Small nitpick, but for the sake of consistency, this should have brackets like the other void functions
```diff
- protected virtual void OnMouseUp(MouseUpEvent e) => Handle(e);
+ protected virtual void OnMouseUp(MouseUpEvent e)
+ {
+     Handle(e);
+ }

File: osu.Framework/Graphics/Drawable.cs [hunk 2]
```diff
@@ -1932,12 +1935,12 @@ protected virtual void OnHoverLost(HoverLostEvent e)
...
         protected virtual bool OnClick(ClickEvent e) => Handle(e);
         protected virtual bool OnDoubleClick(DoubleClickEvent e) => Handle(e);
         protected virtual bool OnDragStart(DragStartEvent e) => Handle(e);
-        protected virtual bool OnDrag(DragEvent e) => Handle(e);
-        protected virtual bool OnDragEnd(DragEndEvent e) => Handle(e);
+        protected virtual void OnDrag(DragEvent e) => Handle(e);
+        protected virtual void OnDragEnd(DragEndEvent e) => Handle(e);
```
Comment (iiSaLMaN) [inline osu.Framework/Graphics/Drawable.cs]: here as well.

File: osu.Framework/Graphics/Drawable.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Framework/Graphics/Drawable.cs]: not refactoring this in this PR.

Comment (smoogipoo) [reply osu.Framework/Graphics/Drawable.cs]: not refactoring this in this PR.

**Pull request merged**